### PR TITLE
Move the references setting back to the DocTopic component

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 109856608

## Summary

Moves the setting of the references back into the `components/DocumentationTopic` instead of the `view`.

This should fix potential rendering issues in Safari.

## Dependencies

NA

## Testing

Ensure all links and interact-able elements on pages are rendered properly and console has no errors.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/DocumentationTopic.vue [hunk 1]
```diff
@@ -622,6 +622,7 @@ export default {
     }
 
     this.store.reset();
+    this.store.setReferences(this.references);
```
Comment (mportiz08) [inline src/components/DocumentationTopic.vue:625]: Sent you more details offline, but we may need to still consider having some way of also doing this as part of the update cycle and not just the initial created hook to support situations where the references may change during the lifetime of the component.

This does seem to fix the rendering issue that we noticed with references in bulleted lists though! Seems like you may have found that there is a race condition with the original way we're setting these refs causing that.

**Comment by dobromir-hristov** [comment]
@swift-ci test

File: src/components/DocumentationTopic.vue [hunk 2]
```diff
@@ -622,6 +622,13 @@ export default {
...
+  },
+  watch: {
+    // update the references in the store, in case they update, but the component is not re-created
+    references(references) {
```
Comment (mportiz08) [inline src/components/DocumentationTopic.vue:629]: Should we also add this watcher to other top-level components?

**Review by mportiz08 — APPROVED** [review]
Seems to be working well in both the initial and updating contexts while still resolving the race condition impacting bulleted lists with links.

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**